batching/batching.go

```diff
@@ -115,19 +116,24 @@ func (b Batches) Iter(ctx context.Context) <-chan data.Entry {
 func (b Batches) Len() int {
 	l := 0
 	for _, batch := range b {
-		l += len(batch)
+		l += len(batch.Data)
 	}
 	return l
 }
 
+type Batch struct {
+	Data
+	age time.Time
+}
+
 // Batch is a map of UIDs to data.
-type Batch map[types.UID]data.Entry
+type Data map[types.UID]data.Entry
 
 func (b *Batch) Map() map[types.UID]data.Entry {
-	if b == nil {
+	if b.Data == nil {
```

Comment:
Seems like we should make this cascading.  

Data should have .Map() and Batch.Map() can call Data.Map().